src/test/java/cloud/fogbow/ras/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -860,83 +902,401 @@ public void testGetUserComputeQuota() throws FogbowException {
+
+        // exercise
+        super.localCloudConnector.deleteInstance(order);
+
+        // verify
+        Mockito.verify(this.publicIpPlugin, Mockito.times(1)).deleteInstance(Mockito.any(),
+                Mockito.any(CloudUser.class));
+    }
+    
+    // test case: When invoking the getAllImages method, the plugin should throw an
+    // InstanceNotFoundException if it does not find the instance.
+    @Test
+    public void testDeleteInstanceThrowsInstanceNotFoundException() throws FogbowException {
+        // set up
+        Order<?> order = Mockito.mock(PublicIpOrder.class);
+        Mockito.when(order.getType()).thenReturn(ResourceType.PUBLIC_IP);
+        Mockito.when(order.getInstanceId()).thenReturn(BaseUnitTests.FAKE_INSTANCE_ID);
+
+        Mockito.doThrow(InstanceNotFoundException.class).when(this.publicIpPlugin).deleteInstance(Mockito.any(),
+                Mockito.any(CloudUser.class));
 
         // exercise
-        localCloudConnector.genericRequest(Mockito.anyString(), systemUser);
+        super.localCloudConnector.deleteInstance(order);
 
         // verify
-        Mockito.verify(systemToCloudMapperPlugin,
-                Mockito.times(1)).map(Mockito.any(SystemUser.class));
-        Mockito.verify(genericRequestPlugin, Mockito.times(1)).
-                redirectGenericRequest(Mockito.any(String.class), Mockito.eq(tokenMock));
+        Mockito.verify(this.publicIpPlugin, Mockito.times(1)).deleteInstance(Mockito.any(),
+                Mockito.any(CloudUser.class));
+    }
+    
+    // test case: When invoking the deleteInstance method of an order with a
+    // different resource type than the COMPUTE, NETWORK, VOLUME, ATTACHMENT,
+    // and PUBLIC_IP, an UnexpectedException will be thrown.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testDeleteInstanceForAnUnsupportedResourceType() throws FogbowException {
+        // set up
+        Order<?> order = Mockito.mock(PublicIpOrder.class);
+        Mockito.when(order.getType()).thenReturn(ResourceType.GENERIC_RESOURCE);
+
+        // exercise
+        super.localCloudConnector.deleteInstance(order);
+    }
+    
+    // test case: When invoking the getAllImages method and the plugin throwing an
+    // UnexpectedException, it must pass this exception.
+    @Test(expected = FogbowException.class) // verify
+    public void testGetAllImagesPassingAnExceptionThrown() throws FogbowException {
+        // set up
+        Mockito.when(this.imagePlugin.getAllImages(Mockito.any(CloudUser.class))).thenThrow(new UnexpectedException());
+
+        SystemUser systemUser = createSystemUser();
+
+        // exercise
+        super.localCloudConnector.getAllImages(systemUser);
+    }
+    
+    // test case: When invoking the getImage method and the plug-in throwing an
+    // UnexpectedException, it must pass this exception.
+    @Test(expected = FogbowException.class) // verify
```

Comment:
same thing

Reply:
The correct is to wait for the release of UnexpectedException, I'll fix this.